code/modules/hallucinations/effects/moderate.dm

```diff
@@ -476,3 +476,48 @@
   */
 /obj/effect/hallucination/delusion/proc/get_image(mob/living/carbon/human/H)
 	return image('icons/mob/animal.dmi', H, pick("black_bear", "brown_bear", "corgi", "cow", "deer", "goat", "goose", "pig", "blank-body"))
+
+// Doppelganger hallucination
+// Spawns a copy of the player that briefly follows them around
+/obj/effect/hallucination/doppelganger
+	duration = 10 SECONDS
+	var/obj/effect/hallucination/chaser/you/fake_you
+
+/obj/effect/hallucination/doppelganger/Initialize(mapload, mob/living/carbon/target)
+	. = ..()
+
+	var/list/locs = list()
+	for(var/turf/T in oview(world.view / 2, target))
+		if(!is_blocked_turf(T))
+			locs += T
+	if(!length(locs))
+		qdel(src)
+		return
+
+	var/turf/T = pick(locs)
+	fake_you = new(T, target)
+
+/obj/effect/hallucination/doppelganger/Destroy()
+	QDEL_NULL(fake_you)
+	return ..()
+
+/obj/effect/hallucination/chaser/you
+	duration = 10 SECONDS
+	min_distance = 2
+	var/image/I = new
+
+/obj/effect/hallucination/chaser/you/Initialize(mapload, mob/living/carbon/target)
+	. = ..()
+	name = "???"
```

Comment:
Why not set the name in the type definition?

Reply:
I believe I did this because all of the other chasers were setting the name in `Initialize`. Let me see if it'll work there.

Reply:
Seems like it has to be on Initialize for chasers? It doesn't work on the type